lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -763,6 +763,18 @@ static void optimizePool(Function *F) {
   } // For all nodes in the graph.
 }
 
+static Variable *getModifiableVariable(Node &node) {
+  // If that node has more than one use, it may not
+  // be okay to modify the underlying variable.
+  if (!node.hasOneUse()) {
+    return nullptr;
+  }
+  if (!isa<Variable>(&node)) {
+    return nullptr;
+  }
+  return cast<Variable>(&node);
```

Comment:
Could replace the isa/cast pair with dyn_cast.  Up to you though, it's pretty clear this way.